includes/REST_API/Stories_Controller.php

```diff
@@ -270,128 +289,137 @@ public function get_items( $request ) {
+				} elseif ( rest_is_object( $tax_include ) ) {
+					$terms            = empty( $tax_include['terms'] ) ? [] : $tax_include['terms'];
+					$include_children = ! empty( $tax_include['include_children'] );
 
-		if ( ! empty( $request['tax_relation'] ) ) {
-			$query_args['tax_query'] = [ 'relation' => $request['tax_relation'] ]; // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
-		}
+					if ( isset( $tax_include['operator'] ) && 'AND' === $tax_include['operator'] ) {
+						$operator = 'AND';
+					}
+				}
 
-		foreach ( $taxonomies as $taxonomy ) {
-			$base        = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;
-			$tax_exclude = $base . '_exclude';
-
-			if ( ! empty( $request[ $base ] ) ) {
-				$query_args['tax_query'][] = [
-					'taxonomy'         => $taxonomy->name,
-					'field'            => 'term_id',
-					'terms'            => $request[ $base ],
-					'include_children' => false,
-				];
+				if ( $terms ) {
+					$args['tax_query'][] = [
+						'taxonomy'         => $taxonomy->name,
+						'field'            => 'term_id',
+						'terms'            => $terms,
+						'include_children' => $include_children,
+						'operator'         => $operator,
+					];
+				}
 			}
 
-			if ( ! empty( $request[ $tax_exclude ] ) ) {
-				$query_args['tax_query'][] = [
-					'taxonomy'         => $taxonomy->name,
-					'field'            => 'term_id',
-					'terms'            => $request[ $tax_exclude ],
-					'include_children' => false,
-					'operator'         => 'NOT IN',
-				];
+			if ( $tax_exclude ) {
+				$terms            = [];
+				$include_children = false;
+
+				if ( rest_is_array( $tax_exclude ) ) {
+					$terms = $tax_exclude;
+				} elseif ( rest_is_object( $tax_exclude ) ) {
+					$terms            = empty( $tax_exclude['terms'] ) ? [] : $tax_exclude['terms'];
+					$include_children = ! empty( $tax_exclude['include_children'] );
+				}
+
+				if ( $terms ) {
+					$args['tax_query'][] = [
+						'taxonomy'         => $taxonomy->name,
+						'field'            => 'term_id',
+						'terms'            => $terms,
+						'include_children' => $include_children,
+						'operator'         => 'NOT IN',
+					];
+				}
 			}
 		}
 
+		return $args;
+	}
+
+	/**
+	 * Add response headers, with post counts.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Response $response Response object.
+	 * @param WP_REST_Request  $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	protected function add_response_headers( WP_REST_Response $response, WP_REST_Request $request ) {
 		// Add counts for other statuses.
 		$statuses = [
-			'all'     => [ 'publish' ],
 			'publish' => 'publish',
 		];
 
 		if ( $this->get_post_type_cap( $this->post_type, 'edit_posts' ) ) {
-			$statuses['all'][]  = 'draft';
-			$statuses['all'][]  = 'future';
 			$statuses['draft']  = 'draft';
 			$statuses['future'] = 'future';
 		}
```

Comment:
Should we use `get_post_stati()` here to make this more generic?

Reply:
How would that work?

Reply:
I don't know, I haven't looked at this in detail.

This just came to mind in #9044.

Instead of hardcoding all possible post statuses here, we might be able to loop through `get_post_stati()` to make this more dynamic and DRY